### ISSUE TITLE
New Camera Icons for Godot Editor: Enhanced Consistency, Modern Design

### ISSUE THREAD

**Issue by Gatada** [godotengine/godot-proposals#8089]
### Describe the project you are working on

Working with camera nodes in the editor.

### Describe the problem or limitation you are having in your project

The current design of the camera icons do not belong: they are inconsistent with most other icons in the editor, and they use an outdated metaphor that I think we should abandon.

Furthermore, having icons with text or captions (I'm look at you XRCamera3D), is not great design.

<img width="248" alt="Screenshot 2023-10-12 at 23 35 40" src="https://github.com/godotengine/godot-proposals/assets/326334/ce9d63d4-2946-4bff-ad00-b8cbda85577f">

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I've designed new camera icons for the Godot editor that offer improved consistency in style and a more modern design that abandons the outdated film reel metaphor, while clearly being a camera.

These icons **seamlessly blend with the existing ones**, ensuring a **cohesive look in the editor** and looking great on both high- and low-DPI screens.

Screenshot showing both 100% and 200% scale, for loDPI and HiDPI screens:

<img width="266" alt="Screenshot 2023-10-12 at 23 55 44" src="https://github.com/godotengine/godot-proposals/assets/326334/6af0b759-2901-47eb-a4b7-b509ade7531f">
<img width="161" alt="Screenshot 2023-10-12 at 23 59 10" src="https://github.com/godotengine/godot-proposals/assets/326334/2f42ee9d-1e38-4045-a188-e29198a9fb18">

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The icons have been made and named to replace the existing icons. It's a simple file replacement operation.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Everyone using the editor will enjoy this update when dealing with camera nodes.

### Is there a reason why this should be core and not an add-on in the asset library?

Editor icons are core, and making the icons more coherent benefits everyone.

**Comment by clayjohn** [comment]
I love the XRCamera3D icon, great job!

**Comment by Mickeon** [comment]
I'm not convinced. The icon being hollow stands out in a way that makes it really similar to Control/Container nodes, implying there's some sort of connection for people
(see also https://github.com/godotengine/godot/pull/81195#issuecomment-1701209748).

I agree that the XRCamera icon could use a change, but so long as it's distinct enough from the original Camera3D while having some semblance to it, it's not that big of a deal.

**Comment by Mickeon** [comment]
Besides, I'm not convinced on the reasoning for the proposal, outright:

> new camera icons for the Godot editor that offer improved consistency in style and a more modern design that abandons the outdated film reel metaphor

It mostly feels like a matter of taste. There's nothing wrong with an "outdated metaphor" by itself. The most common example in all software is usage of a floppy disk to symbolise saving to disk. 
Godot is no stranger to it, either. If anything, a few outdated metaphors may fit more with Godot's overall style:
- Almost all Animation classes' icons bear a strong resemblance to film reels or tapes;
![image](https://github.com/godotengine/godot-proposals/assets/66727710/41771229-6c61-4357-af22-2f98af05aa11)
![image](https://github.com/godotengine/godot-proposals/assets/66727710/a328d329-6f07-4890-85c3-ace71f83ebb2)
- The Script icon looks like an... actual script for a theatrical performance;
![image](https://github.com/godotengine/godot-proposals/assets/66727710/578cdf23-1324-4b76-a7e0-e8896e74648c)
- A majority of Texture icons resemble pictures taken from an analog photo camera;
![image](https://github.com/godotengine/godot-proposals/assets/66727710/a1373629-bf91-4344-8485-352ea182507c)

The association to real objects makes them easier to visualise. There's probably a few more subtle cases.

**Comment by MewPurPur** [comment]
Overall agree. I especially like them when filled up. I don't like the XR icon though, I feel like the extra line makes it look not much like a camera. My opinion is that we need a little discerning symbol for XR classes that's more distinct, like with CSG classes (maybe bottom left?) and to apply it to all of them. But that's another discussion

**Comment by YuriSizov** [comment]
I mentioned this on chat already when these were discussed, but I think that the proposed shapes go too far into the abstract territory. They are well designed, but they are comparatively harder to parse as instead of one camera-like shape you have a bunch of geometric forms.

That makes @Mickeon's suggestion above to make them filled so much better in my opinion. Especially the XR camera one. But they still lose their uniqueness and become rather bland. I think we can keep the base camera icon as is, but we can explore a design similar to the proposed one for the XR camera. The idea to reference VR headset is very nice.

**Comment by passivestar** [comment]
I personally don't mind icons using outdated metaphors, but the proposed icon is so clean, I love it! Especially the filled one. It could use a tiny bit more separation between the two parts though imo, so that it doesn't merge as much into a single blob from the distance:
<img width="282" alt="image" src="https://github.com/godotengine/godot-proposals/assets/60579014/d2be1a3f-301c-43d0-8d3f-6ec22d6afacb">

And for the XR one I think a symbol is better like @MewPurPur suggested, otherwise it's hard to understand

**Comment by Gatada** [comment]
<img width="543" alt="Screenshot 2023-10-19 at 08 18 51" src="https://github.com/godotengine/godot-proposals/assets/326334/8bc84504-0ea8-4154-a2a9-451746a8b0f3">

Using my initial outline design, with 130 votes, Twitter.. ugh.. I mean X, deemed no update as the best option.

Does that kill this proposal, or should I rework the icons with fill and a common (non-typographical) XR element?

**Comment by Mickeon** [comment]
A single poll like that does not "kill" the proposal. Polls may be skewed for a multitude or reasons. It does give the proposal less "confidence", though. As in, you are now aware that a majority of people _(not even necessarily editor users)_ saw the poll and prefer the current design.

**Comment by MewPurPur** [comment]
I agree, maybe the results would be different if you suggested the filled in camera for example
